idx/memory/memory.go

```diff
@@ -265,7 +265,8 @@ func (m *MemoryIdx) Load(defs []schema.MetricDefinition) int {
 }
 
 func (m *MemoryIdx) add(def *schema.MetricDefinition) idx.Archive {
-	path := def.Name
+	path := def.FullNameWithTags()
```

Comment:
It seems wasteful to save an extra copy of everything when realistically the `Name` and `Tags` could all reference the one string.

Reply:
very good idea, going to try that

Reply:
that's a good point, i wonder if it makes sense to use 1 backing string and either parse out substrings on-demand, or maintain a list of index integers.

Reply:
Hmm, that's true. Could trim off the half the slice size because we don't need to store a pointer, just a start and end.

Reply:
The problem is that if we would store name and tags in only one string and then keep the positions where name&tags start/end as integers, then the `Name` could not be accessible as a simple property anymore unless we make the tags part of its value. We could easily add methods `Name()`  and `Tags()` to extract those substrings, but that would then change the interface. 
So I think probably the best would be to use one string with everything and then slice it up into names & tags, even though that's producing more pointers

Reply:
Something like that? or did i make it unnecessarily complicated?
https://github.com/raintank/schema/pull/12/commits/bc847ce79f0deb7b03ecaa983a1bd1115f6d5c4e

Reply:
>  then the Name could not be accessible as a simple property anymore unless we make the tags part of its value. We could easily add methods Name() and Tags() to extract those substrings, but that would then change the interface.

This is not a problem if we're confident that it is results in significant performance improvement.
That said i'm not convinced that at this point it's worth the time spend to figure out whether or not it would result in a significant performance improvement. We can always push that for later.
One thing I would add though is that if a function/method implementation is simple enough (e.g. nothing more than returning a property value or maybe a few lines more) than the compiler will just inline that function and there is no function call overhead. but it can only do this when the type is fixed at compile time (not an interface type)

Reply:
> Something like that? or did i make it unnecessarily complicated?
raintank/schema@bc847ce

it looks more complicated because a var was renamed from buffer to idBuffer. if it kept the name it would be more easy to follow. otherwise it looks pretty good to me

Reply:
@Dieterbe I updated it again and moved the deduplication into a separate function. That way it can also be used in other places, like for example after loading the definitions from the cassandra index, without side effects.